cirq-core/cirq/ops/raw_types.py

```diff
@@ -688,41 +688,7 @@ def _commutes_(
         """Determine if this Operation commutes with the object"""
         if not isinstance(other, Operation):
             return NotImplemented
-
-        self_keys = protocols.measurement_key_objs(self)
-        other_keys = protocols.measurement_key_objs(other)
-        if (
-            not self_keys.isdisjoint(other_keys)
-            or not protocols.control_keys(self).isdisjoint(other_keys)
-            or not protocols.control_keys(other).isdisjoint(self_keys)
-        ):
-            return False
-
-        if hasattr(other, 'qubits') and set(self.qubits).isdisjoint(other.qubits):
-            return True
-
-        from cirq import circuits
-
-        # Remove the classical controls to validate the quantum commutativity. This can be done
-        # because during execution, the two operations will either both be run, in which case they
-        # behave like the suboperations, so if the suboperations commute then these commute. Or
-        # one of them is cold in which case it behaves like the identity, which always commutes.
-        self_raw = self.without_classical_controls()
-        other_raw = other.without_classical_controls()
-        circuit12 = circuits.Circuit(self_raw, other_raw)
-        circuit21 = circuits.Circuit(other_raw, self_raw)
-
-        # Don't create gigantic matrices.
-        shape = protocols.qid_shape_protocol.qid_shape(circuit12)
-        if np.prod(shape, dtype=np.int64) > 2**10:
-            return NotImplemented  # pragma: no cover
-
-        m12 = protocols.unitary_protocol.unitary(circuit12, default=None)
-        m21 = protocols.unitary_protocol.unitary(circuit21, default=None)
-        if m12 is None or m21 is None:
-            return NotImplemented
-
-        return np.allclose(m12, m21, atol=atol)
+        return _operations_commutes_impl([self], [other], atol=atol)
```

Comment:
Should this also check commutes(other, self) for consistency?

Reply:
AFAICT, the `_operations_commutes_impl` is symmetric in its arguments.
The `protocol.commutes` will also do it for us [anyway](https://github.com/chamodtharakaperera/Cirq/blob/0551f85c5beb26e17187f6f51e3cfd6536ac1327/cirq-core/cirq/protocols/commutes_protocol.py#L160-L174).